FEXCore/Source/Interface/Core/OpcodeDispatcher.cpp

```diff
@@ -4176,35 +4176,41 @@ AddressMode OpDispatchBuilder::SelectAddressMode(AddressMode A, bool AtomicTSO,
   // addresses are reserved and therefore wrap around is invalid.
   //
   // TODO: Also handle GPR TSO if we can guarantee the constant inlines.
-  if (SupportsRegIndex) {
-    if ((A.Base || A.Segment) && A.Offset) {
-      const bool Const_16K = A.Offset > -16384 && A.Offset < 16384 && A.AddrSize == OpSize::i32Bit && GPRSize == OpSize::i32Bit;
-
-      if ((A.AddrSize == OpSize::i64Bit) || Const_16K) {
-        // Peel off the offset
-        AddressMode B = A;
-        B.Offset = 0;
-
-        return {
-          .Base = LoadEffectiveAddress(B, true /* AddSegmentBase */, false),
-          .Index = _Constant(A.Offset),
-          .IndexType = MEM_OFFSET_SXTX,
-          .IndexScale = 1,
-        };
-      }
+  if ((A.Base || A.Segment) && A.Offset) {
+    const bool SupportsSmallConst = A.AddrSize == OpSize::i32Bit && GPRSize == OpSize::i32Bit;
```

Comment:
I think this needs to be `const bool SupportsSmallConst = A.AddrSize ==  GPRSize;` Otherwise we lose small const support for LRCPC2 on 64-bit processes with 64-bit addresses?

Reply:
I think the small const logic is wacked as it is on 64-bit. I'll investigate this a bit more.